src/dist/dist.rs

```diff
@@ -614,6 +614,15 @@ fn update_from_dist_<'a>(
     let mut fetched = String::new();
     let mut first_err = None;
     let backtrack = toolchain.channel == "nightly" && toolchain.date.is_none();
+    // We want to limit backtracking if we do not already have a toolchain
+    let mut backtrack_limit: Option<i32> = if old_date.is_some() {
+        None
+    } else {
+        // The limit of 20 here is an arbitrary selection, let the user override it.
+        let provided = env::var("RUSTUP_BACKTRACK_LIMIT").unwrap_or_else(|_| "20".to_owned());
```

Comment:
This is a super-nit, but I'd be inclined to set the default to either a full release cycle (6w) or a half release cycle (3w). Still arbitrary, but feels a little less arbitrary than 20. We're essentially here trading "chance of success" against "time to get an error if you misspell a component", where I think the chance of success has diminishing returns at anywhere over 2w.

Reply:
Since 20 is close to 3w, I've updated it to 21 and state it's half a release cycle.